osf/models/preprint_provider.py

```diff
@@ -21,6 +21,7 @@ class PreprintProvider(ObjectIDMixin, BaseModel):
     header_text = models.TextField(default='', blank=True)
     description = models.CharField(null=True, blank=True, max_length=256)  # max length on prod: 56
     banner_name = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 19
+    domain = models.CharField(null=True, blank=True, max_length=253)  #
```

Comment:
If this will be a fully qualified url django has a [URLField](https://docs.djangoproject.com/en/1.11/ref/models/fields/#urlfield) that will do validation for you.